radicle-httpd/src/api/v1/projects.rs

```diff
@@ -230,6 +238,47 @@ async fn commit_handler(
     Ok::<_, Error>(Json(response))
 }
 
+/// Get diff between two commits
+/// `GET /projects/:project/diff/:base/:oid`
+async fn diff_handler(
+    State(ctx): State<Context>,
+    Path((project, base, oid)): Path<(Id, Oid, Oid)>,
+    Query(qs): Query<DiffQueryString>,
+) -> impl IntoResponse {
+    let DiffQueryString { page, per_page } = qs;
+    let storage = &ctx.profile.storage;
+    let repo = Repository::open(paths::repository(storage, &project))?;
+    let base = repo.commit(base)?;
+    let commit = repo.commit(oid)?;
+    let diff = repo.diff(base.id, commit.id)?;
+
+    // If a pagination is defined, we do not want to paginate the commits, and we return all of them on the first page.
+    let page = page.unwrap_or(0);
```

Comment:
It seems weird that we would paginate the commits, but return the full diff. Could we just remove pagination for now? Or do we think we need it?

Reply:
I wanted to have a similar commit listing as GH, on their PRs and the diff to be the full one, since we would show them in different tabs.
But I think we can allow us to not have commit pagination for now.

Reply:
Ah is this in the "compare" view? I didn't realize they paginated commits there.

Reply:
Yes I thought about adding a commit listing like GitHub does under the commit tab
<img width="694" alt="Bildschirm­foto 2023-03-21 um 13 14 32" src="https://user-images.githubusercontent.com/7912302/226602791-bb6db9cb-6917-44f2-98d6-5041eb01eb34.png">

But I finally found a big PR and it seems they don't paginate..
https://github.com/torvalds/linux/pull/797/commits

So I'll leave the commit pagination out for now

Reply:
Aha but they also don't show all commits! " This pull request is big! We’re only showing the most recent 250 commits. "

So yeah, let's return all commits and deal with the case of "too many commits" if it comes up.